jaxrs-testing/src/test/java/com/facebook/airlift/jaxrs/testing/TestThriftResponseHandler.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs.testing;
+
+import com.facebook.airlift.http.client.HeaderName;
+import com.facebook.airlift.http.client.Request;
+import com.facebook.airlift.http.client.Response;
+import com.facebook.airlift.http.client.ResponseHandler;
+import com.facebook.airlift.jaxrs.thrift.TInputStreamTransport;
+import com.facebook.drift.codec.ThriftCodec;
+import com.facebook.drift.protocol.TFacebookCompactProtocol;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ListMultimap;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.List;
+
+import static com.facebook.airlift.http.client.ResponseHandlerUtils.propagate;
+
+public class TestThriftResponseHandler<T>
+        implements ResponseHandler<TestThriftResponseHandler.ThriftResponse, RuntimeException>
+{
+    private final ThriftCodec<T> thriftCodec;
+
+    public TestThriftResponseHandler(ThriftCodec<T> thriftCodec)
+    {
+        this.thriftCodec = thriftCodec;
+    }
+
+    @Override
+    public ThriftResponse handleException(Request request, Exception exception)
+    {
+        throw propagate(request, exception);
+    }
+
+    @Override
+    public ThriftResponse handle(Request request, Response response)
+    {
+        try {
+            return new ThriftResponse(response.getStatusCode(), response.getStatusMessage(), response.getHeaders(), thriftCodec, response.getInputStream());
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public class ThriftResponse
+    {
+        private final int statusCode;
+        private final String statusMessage;
+        private final ListMultimap<HeaderName, String> headers;
+        private final T value;
+
+        ThriftResponse(int statusCode, String statusMessage, ListMultimap<HeaderName, String> headers, ThriftCodec<T> thriftCodec, InputStream thriftBytes)
+        {
+            this.statusCode = statusCode;
+            this.statusMessage = statusMessage;
+            this.headers = ImmutableListMultimap.copyOf(headers);
+            T value = null;
+            IllegalArgumentException exception = null;
+            try {
+                value = thriftCodec.read(new TFacebookCompactProtocol(new TInputStreamTransport(thriftBytes)));
```

Comment:
Let's move this logic out of constructor and pass here the value

Reply:
It feels like `TRANSPORT` and `PROTOCOL` should be auto discoverable.

Reply:
See how it is done in the thrift RPC protocol: https://github.com/prestodb/drift/blob/master/drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/server/ThriftProtocolDetection.java#L73

I think also the other direction worth exploring - if we can encode protocol and transport into the Content-Type header. For example we can use content type directives to define transport and protocol:

e.g.: `Content-Type: application/x-thrift; tp={u|f|h}.{b|c|f}`

sample: `Content-Type: application/x-thrift; tp=fb` - framed, binary
sample: `Content-Type: application/x-thrift; tp=hf` - header, fbcompact

To easy manipulate the content type string the `javax.ws.rs.core.MediaType` class can be used